pkg/virt-launcher/notify-client/client.go

```diff
@@ -151,19 +155,32 @@ func libvirtEventCallback(c cli.Connection, domain *api.Domain, libvirtEvent lib
 	}
 }
 
-func (c *NotifyClient) StartDomainNotifier(domainConn cli.Connection, deleteNotificationSent chan watch.Event, vmiUID types.UID) error {
-
+func (c *NotifyClient) StartDomainNotifier(domainConn cli.Connection, deleteNotificationSent chan watch.Event, vmiUID types.UID, qemuAgentPollerInterval *time.Duration) error {
 	eventChan := make(chan libvirtEvent, 10)
 	reconnectChan := make(chan bool, 10)
 
 	domainConn.SetReconnectChan(reconnectChan)
 
+	qemuAgentCallback := func(domainName string, interfaceStatuses *[]api.InterfaceStatus) {
+		eventCallback(domainConn, util.NewDomainFromName(domainName, vmiUID), libvirtEvent{}, c, deleteNotificationSent, interfaceStatuses)
+	}
+	agentPoller := agentpoller.CreatePoller(domainConn, vmiUID, qemuAgentCallback, qemuAgentPollerInterval)
```

Comment:
Can you send back an event with the changed data, if it changed, instead of the callback?